universal-application-tool-0.0.1/app/modules/SecurityModule.java

```diff
@@ -156,17 +157,51 @@ protected OidcClient provideAdClient(
       return null;
     }
     OidcConfiguration config = new OidcConfiguration();
+    // this is the resource identifier that tells adfs that this is civiform
     config.setClientId(this.configuration.getString("adfs.client_id"));
+
+    // this is the token that we create within adfs
     config.setSecret(this.configuration.getString("adfs.secret"));
+
+    // oidc convention that auth provider has an endpoint that can get text blob
+    // that gets back public key/etc (can just curl this if curious,
+    // get from endpoints in azure)
     config.setDiscoveryURI(this.configuration.getString("adfs.discovery_uri"));
+
+    // tells AD to use a post response when it sends info back from
+    // the auth request
     config.setResponseMode("form_post");
+
+    // we want adfs to give us an id token back from this request
     config.setResponseType("id_token");
-    config.setScope("openid profile email allatclaims");
+
+    // scopes are the other things that we want from the adfs endpoint
+    // (needs to also be configured on adfs).
+    // note azure ad has the extra claim: allatclaims which returns
+    // access token in the id_token. might be used for
+    // the group that tells who are in what group.
+    String[] defaultScopes = {"openid", "profile", "email"};
+    String[] extraScopes = this.configuration.getString("adfs.additional_scopes").split(" ");
+    ArrayList<String> allClaims = new ArrayList<>();
+    Collections.addAll(allClaims, defaultScopes);
+    Collections.addAll(allClaims, extraScopes);
+    config.setScope(String.join(" ", allClaims));
+
+    // security setting that adds a random number to ensure cannot be reused
     config.setUseNonce(true);
+
+    // ? relationship with adfs is not stateful ?
```

Comment:
Can you reword into a todo 

EG:

// TODO: Investigate the need for this.

Reply:
added what it does! allows for custom state values to be defined